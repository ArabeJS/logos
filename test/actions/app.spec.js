import { goTo, showAlert, hideAlert, login, logOut } from 'actions';

describe('App', () => {
  it('goTo should create an action to navigate with react-router', () => {
    expect([goTo('/destination')]).toMatchSnapshot();
  });

  it('hideAlert should return an action', () => {
    expect(hideAlert('ABC1')).toEqual({
      type: 'HIDE_ALERT',
      payload: { id: 'ABC1' },
    });
  });

  it('showAlert should return an action', () => {
    expect(showAlert('Alright!', { type: 'success', id: 'ABC1' })).toEqual({
      type: 'SHOW_ALERT',
      payload: {
        id: 'ABC1',
        message: 'Alright!',
        position: 'bottom-right',
        type: 'success',
        timeout: 5,
      },
    });

    expect(showAlert('ERROR!', { type: 'error', id: 'ABC2', timeout: 0, position: 'top' })).toEqual({
      type: 'SHOW_ALERT',
      payload: {
        id: 'ABC2',
        message: 'ERROR!',
        position: 'top',
        type: 'error',
        timeout: 0,
      },
    });
  });

  it('login should return an action', () => {
    expect(login()).toMatchSnapshot();
  });

  it('logOut should return an action', () => {
    expect(logOut()).toMatchSnapshot();
  });
});
