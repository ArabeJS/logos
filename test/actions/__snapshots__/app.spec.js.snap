// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App goTo should create an action to navigate with react-router 1`] = `
Array [
  Object {
    "payload": Object {
      "args": Array [
        Object {
          "pathname": "/destination",
          "search": undefined,
          "state": undefined,
        },
      ],
      "method": "push",
    },
    "type": "@@router/CALL_HISTORY_METHOD",
  },
]
`;

exports[`App logOut should return an action 1`] = `
Object {
  "payload": Object {},
  "type": "USER_LOGOUT_REQUEST",
}
`;

exports[`App login should return an action 1`] = `
Object {
  "payload": Object {
    "provider": undefined,
  },
  "type": "USER_LOGIN_REQUEST",
}
`;
